### PR TITLE
IGNITE-18156 Sql. Extend SQL grammar with CREATE/DROP ZONE statement.

### PR DESCRIPTION
**Pull request by xtern**
Basic SQL grammar to CREATE/DROP distributed zones.

It was decided not to overload the parser with complex validation logic of command options (it should be implemented somewhere near the conversion level).

Thus, currently the parser is not responsible for invalid (unknown) options, invalid option invariants, and duplicates.

### CODE REVIEW

File: modules/sql-engine/src/main/codegen/config.fmpp [hunk 1]
```diff
@@ -53,6 +56,7 @@ data: {
 #     "KEY_TYPE" // already presented in Calcite
       "TREE"
       "HASH"
+#      "ZONE"
```
Comment (ygerzhedovich) [inline modules/sql-engine/src/main/codegen/config.fmpp:59]: why it commented?

Comment (xtern) [reply modules/sql-engine/src/main/codegen/config.fmpp]: Thanks, I forgot to remove it from this list, It is not reserved keyword in Calcite.

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/sql/IgniteSqlCreateZone.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+        if (createOptionList != null) {
+            writer.keyword("WITH");
+
+            createOptionList.unparse(writer, 0, 0);
+        }
+    }
+
+    /**
+     * Get name of the distributed zone.
```
Comment (korlov42) [inline modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/sql/IgniteSqlCreateZone.java:83]:
```suggestion
     * Get name of the distribution zone.
```

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/sql/IgniteSqlCreateZone.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+     */
+    public SqlIdentifier name() {
+        return name;
+    }
+
+    /**
+     * Get list of the specified options to create distributed zone with.
```
Comment (korlov42) [inline modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/sql/IgniteSqlCreateZone.java:90]:
```suggestion
     * Get list of the specified options to create distribution zone with.
```

File: modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/sql/SqlDdlZoneParserTest.java [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+
+/**
+ * Test suite to verify parsing of the DDL "ZONE" commands.
+ */
+public class SqlDdlZoneParserTest extends AbstractDdlParserTest {
+    /**
+     * Parse simple CREATE ZONE statement.
+     */
+    @Test
+    public void createZoneMoOptions() throws SqlParseException {
```
Comment (korlov42) [inline modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/sql/SqlDdlZoneParserTest.java:47]:
```suggestion
    public void createZoneNoOptions() throws SqlParseException {
```

File: modules/sql-engine/src/main/codegen/includes/parserImpls.ftl [hunk 5]
```diff
@@ -161,6 +161,65 @@ SqlNodeList TableElementList() :
     }
 }
 
+SqlCreate SqlCreateZone(Span s, boolean replace) :
```
[the remark this answers is not in the document]
Comment (xtern) [reply modules/sql-engine/src/main/codegen/includes/parserImpls.ftl]: Done

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/sql/IgniteSqlCreateZone.java (hunk 2 above)
Comment (xtern) [reply modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/sql/IgniteSqlCreateZone.java]: Fixed, thanks.

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/sql/IgniteSqlCreateZone.java (hunk 3 above)
Comment (xtern) [reply modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/sql/IgniteSqlCreateZone.java]: Fixed, thanks.

File: modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/sql/SqlDdlZoneParserTest.java (hunk 4 above)
Comment (xtern) [reply modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/sql/SqlDdlZoneParserTest.java]: Fixed, thanks.

File: modules/sql-engine/src/main/codegen/includes/parserImpls.ftl [hunk 6]
```diff
@@ -161,6 +161,65 @@ SqlNodeList TableElementList() :
...
+}
+
+void CreateZoneOption(List<SqlNode> list) :
+{
+    final Span s;
+    final SqlIdentifier key;
+    final SqlNode val;
+}
+{
+    key = SimpleIdentifier() { s = span(); }
```
[the remark this answers is not in the document]
Comment (xtern) [reply modules/sql-engine/src/main/codegen/includes/parserImpls.ftl]: Done.

**Review by korlov42 — APPROVED** [review]

**Pull request merged**